### ISSUE TITLE
App/database access should support per-session MFA

### ISSUE THREAD

**Issue by r0mant** [gravitational/teleport#6172]
### What

6.1 release introduced per-session MFA with which Teleport can prompt users to tap their registered key for each established SSH session or each executed K8s command:

https://www.youtube.com/watch?v=74g0V2XwbE4

With `tsh ssh` we can do that because we control the client, however with app/db access clients such as `psql` or `curl` we don't control them and they don't know to prompt the user to tap their key. Couple of quick ideas briefly discussed before:

* Have some client-side forward proxy all clients would talk to.
* Have `tsh psql`, `tsh mysql` and other similar aliases.
* Do some OS level magic with detecting that certificate file is accessed and prompting the user to tap the key.

We need to do a proper research, come up with the design and implement per-session MFA support for app/db access.

**Comment by zmb3** [comment]
@r0mant can this one be closed? Pretty sure we now support MFA for apps and dbs.

**Comment by webvictim** [comment]
@zmb3 I don't think we do, at least for apps. I tried to access an app with per-session MFA enabled the other day and it didn't work at all in 9.0.4. I added a comment to this issue: https://github.com/gravitational/teleport/issues/8466

This could theoretically be quite easy to enable at the browser level, as we could require the MFA tap before the session starts, similar to how it's implemented for desktop access. It wouldn't work at the command line, but I think a high proportion of our application access use cases are just HTTP reverse proxying in a browser.

**Comment by russjones** [comment]
For Database Access, support to per-session MFA authenticate the entire connection via either `tsh proxy db` or Teleport Connect will be coming out in early March.

For Application Access, we don't have an ETA at the moment for adding support for per-session MFA.

**Comment by zmb3** [comment]
See also #6172

**Comment by snags141** [comment]
Any updates on where this sits on the roadmap, or planned/existing workarounds?

**Comment by hrdflt** [comment]
Second this request -- Any updates here? What is the workaround? This functionality is supported by competitor products like `pomerium` (device identity enforcement for web apps).
